ilastik/utility/data_url.py

```diff
@@ -0,0 +1,522 @@
+from abc import ABC, abstractmethod
+from typing import ClassVar, Dict, Mapping, Set, Tuple, TypeVar, Sequence, Optional, List, Type, Iterable, Union, cast
+from pathlib import PurePosixPath, Path
+import errno
+import glob
+import os
+import re
+from urllib.parse import urlencode, urlsplit, parse_qs
+import enum
+
+import numpy as np
+from typing_extensions import final
+import z5py
+import h5py
+
+from lazyflow.utility.pathHelpers import lsH5N5, globH5N5, globNpz
+
+# pyright: strict
```

Comment:
looks like you introduce new tooling here. Is it comparable to mypy? Is it better?

Reply:
It's... complementary to mypy, I guess. It does some things better, some things worse. Having that comment there shouldn't hurt anyone, but I can mark that file to be strictly checked elsewhere (outside of the source code) if it's bothering you =)

Reply:
Do you check other parts of the code without the strict option? Or did you only check this one file at all? When would you have non-strict checks (strict sounds like the right thing to do? :D)...

In any case, maybe that is better suited for `pyproject.toml`?